src/lib/web-component/filter-option/filter-option-element.js

```diff
@@ -27,32 +30,26 @@ class FilterOption extends HTMLElement {
   updateFilterCheckboxes(logs) {
     for (const log of logs) {
       const checkboxLabel = log[this.filterKey];
-      if (!this.checkboxLabels.has(checkboxLabel)) {
-        this.checkboxLabels.add(checkboxLabel);
+      if (!this.viewCheckboxLabels.has(checkboxLabel)) {
+        this.viewCheckboxLabels.add(checkboxLabel);
         this.addNewCheckbox(checkboxLabel);
-
-        const filterDetail = {
-          filterObject: { logKey: this.filterKey, valueEquals: checkboxLabel },
-          isNewFilterOption: true,
-        };
-        this.dispatchEvent(
-          new CustomEvent('filterchange', {
-            detail: filterDetail,
-          })
-        );
+        this.dispatchFilterChangeEvent({ isNewFilterAdded: true });
       }
     }
   }
 
-  addNewCheckbox(labelText) {
+  addNewCheckbox(checkboxLabel) {
     const newCheckbox = this.checkboxTemplate.cloneNode(true);
     newCheckbox.querySelector('label span').textContent =
-      labelText || 'undefined';
+      checkboxLabel || 'undefined';
 
     const inputCheckbox = newCheckbox.querySelector('input');
-    inputCheckbox.value = labelText;
+    inputCheckbox.value = checkboxLabel;
     inputCheckbox.checked = true;
 
+    // adding to activeCheckboxLabels list since checkbox is checked
```

Comment:
Nice. This is an example of a good comment. It explains why the code is like this, when it is not immediately obvious from the code.